superduperdb/components/datatype.py

```diff
@@ -380,11 +392,27 @@ def decode(cls, r, db=None) -> 'Encodable':
         object = cls._get_object(db, r['_content'])
         return cls(
             x=object,
-            datatype=db.datatypes[r['_content']['datatype']],
+            datatype=cls.get_datatype(db, r['_content']),
             uri=r['_content']['uri'],
             file_id=r['_content'].get('file_id'),
         )
 
+    @classmethod
+    def get_datatype(cls, db, r):
```

Comment:
can we add docstrings in extra added function in this file?

Reply:
Done